projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts

```diff
@@ -65,11 +68,24 @@ export class IonDatePickerCalendarComponent implements OnInit, OnChanges {
   };
   public finalRange = true;
 
-  constructor() {
-    this.setLanguage();
+  previousYear(): void {
+    this.calendar.goToPreviousYear(this.calendar.month.number - 1);
+  }
+
+  previousMonth(): void {
+    this.calendar.goToPreviousMonth();
+  }
+
+  nextMonth(): void {
+    this.calendar.goToNextMonth();
+  }
+
+  nextYear(): void {
+    this.calendar.goToNextYear(this.calendar.month.number - 1);
   }
 
   ngOnInit(): void {
+    arrangeDates(this.selectedDay);
```

Comment:
```suggestion
    arrangeDates(this.selectedDay);
```

Check length this.selectedDay before